fury/stream/client.py

```diff
@@ -0,0 +1,362 @@
+import vtk
+import sys
+import logging
+import time
+if sys.version_info.minor >= 8:
+    PY_VERSION_8 = True
+else:
+    shared_memory = None
+    PY_VERSION_8 = False
+
+import numpy as np
+from functools import partial
+from fury.stream.tools import ArrayCircularQueue, SharedMemCircularQueue
+from fury.stream.tools import (
+    RawArrayImageBufferManager, SharedMemImageBufferManager)
+from fury.stream.tools import IntervalTimer
+from fury.stream.constants import _CQUEUE
+
+
+def callback_stream_client(*args, **kwargs):
+    """This callback is used to update the image inside of
+    the ImageManager instance
+
+    Parameters
+    ----------
+    args : args
+    kwargs : kwargs
+
+    """
+    stream_client = kwargs['stream_client']
+    if not stream_client._in_request:
+        stream_client._in_request = True
+        stream_client.window2image_filter.Update()
+        stream_client.window2image_filter.Modified()
+        vtk_image = stream_client.window2image_filter.GetOutput()
+        vtk_array = vtk_image.GetPointData().GetScalars()
+        # num_components = vtk_array.GetNumberOfComponents()
+
+        w, h, _ = vtk_image.GetDimensions()
+        np_arr = np.frombuffer(vtk_array, dtype='uint8')
+        if np_arr is not None:
+            stream_client.img_manager.write_into(w, h, np_arr)
+            # time.sleep(kwargs['ms']/1000)
+            stream_client._in_request = False
+            if kwargs['force_render']:
+                stream_client.showm.window.Render()
+                stream_client.showm.iren.Render()
```

Comment:
in python, to avoid unnecessary indentation, you should inverse the logic,  see below:
```python
if stream_client._in_request:
    return
stream_client._in_request = True
stream_client.window2image_filter.Update()
.....
if np_arr is None:
   return
stream_client.img_manager.write_into(w, h, np_arr)
....
```